### PR TITLE
Wrapped both edge aircraft detection containers under the edgetech-core paradigm and merged into SkyScan 

### PR DESCRIPTION
**Pull request by meadej**

### CODE REVIEW

File: .github/workflows/dockerbuild.yml [hunk 1]
```diff
@@ -39,8 +39,9 @@ jobs:
     - name: Build Images
       env: 
         # Every folder in the repo that has a Dockerfile within it, comma separated 
-        DOCKERFOLDERS: "edge-detect/ai,tracker,piaware,pan-tilt-pi,notebook-server,mqtt,egi,axis-ptz"
+        DOCKERFOLDERS: "edge-detect/coral,edge-detect/yolo,axis-ptz,egi,mqtt,notebook-server,piaware,tracker"
         PROJECT_NAME: "skyscan"
+        REPO_NAME: "meadejs"
```
Comment (mchadwick-iqt) [inline .github/workflows/dockerbuild.yml]: I think this needs to be changed to the correct name

File: edge-detect/coral/base_mqtt_pub_sub.py [hunk 2]
```diff
@@ -0,0 +1,319 @@
+"""This file include the parent class for all future edge technology work by IQT Labs that
```
Comment (mchadwick-iqt) [inline edge-detect/coral/base_mqtt_pub_sub.py]: @meadej - Is there a way to grab this file in the docker build process from the edgetech-core repo?  I'd like to not have a copy of this file outside of its source control

File: edge-detect/coral/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+version: '3.7'
+
+services:
+  coral-edge:
+    image: skyscan_fast_ai_image:latest
+    privileged: true
+    volumes:
+      - /dev/bus/usb:/dev/bus/usb
+      - /home/mobian/dphogan_test/edge_test/tosort:/data/tosort
```
Comment (mchadwick-iqt) [inline edge-detect/coral/docker-compose.yml]: If this file is needed, it should include paths that will exist

File: edge-detect/coral/docker_setup.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
```
Comment (mchadwick-iqt) [inline edge-detect/coral/docker_setup.sh]: Is this file necessary?

File: edge-detect/coral/sort_coral.py [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+                    print('noplane')
+                    shutil.move(img_path, os.path.join(
+                        self.noplane_dir, os.path.basename(img_path)))
+
+                # Write JSON
+                if self.save_json and count > 0:
+                    struct = []
+                    for obj in objs:
+                        if obj.id == self.goal_index:
+                            struct.append({
```
Comment (mchadwick-iqt) [inline edge-detect/coral/sort_coral.py]: @meadej  @dphogan - I think it makes sense to append some information about the model that ran the bounding box.  Can you add something in here with model type (TFLite), and some kind of version of the model in use?  Not sure what specifically that looks like

File: edge-detect/docker-compose.yml [hunk 6]
```diff
@@ -3,13 +3,13 @@ version: '3.7'
 services:
```
Comment (mchadwick-iqt) [inline edge-detect/docker-compose.yml]: Is this file needed?

File: edge-detect/yolo/base_mqtt_pub_sub.py [hunk 7]
```diff
@@ -0,0 +1,319 @@
+"""This file include the parent class for all future edge technology work by IQT Labs that
```
Comment (mchadwick-iqt) [inline edge-detect/yolo/base_mqtt_pub_sub.py]: @meadej Can this be pulled from the source rather than copied?  Same comment as before, I would like this to come from source control

File: edge-detect/yolo/Dockerfile.old [hunk 8]
```diff
@@ -0,0 +1,41 @@
+FROM iqtlabs/edgetech-core:latest
```
Comment (mchadwick-iqt) [inline edge-detect/yolo/Dockerfile.old]: @meadej is this file needed?

File: edge-detect/yolo/sort.py [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+                print('plane')
+                shutil.move(path, os.path.join(self.plane_dir, os.path.basename(path)))
+            else:
+                shutil.move(path, os.path.join(self.noplane_dir, os.path.basename(path)))
+                print('noplane')
+
+            if self.save_json:
+                struct = []
+                for rowidx in range(pred[0].shape[0]):
+                    row = pred[0][rowidx, :]
```
Comment (mchadwick-iqt) [inline edge-detect/yolo/sort.py]: @meadej  @dphogan - I think it makes sense to append some information about the model that ran the bounding box.  Can you add something in here with model type (yolo), and some kind of version of the model in use?  Not sure what specifically that looks like

File: .github/workflows/dockerbuild.yml [hunk 10]
```diff
@@ -52,7 +53,7 @@ jobs:
           echo "Building $folder"
```
Comment (mchadwick-iqt) [inline .github/workflows/dockerbuild.yml]: wont SUBNAME=${NAMEFOLDER[0]} name the container image "edge-detect" instead of something like "edge-detect-yolo"?  Will one of the images overwrite the other during the CI process?

File: axis-ptz/camera.py [hunk 11]
```diff
@@ -81,15 +81,46 @@
 
 currentPlane = None
 
-camera_latitude = None
```
Comment (mchadwick-iqt) [inline axis-ptz/camera.py]: I may be mistaken, but aren't these needed to initialize the global variables?

File: docker-compose.yml [hunk 12]
```diff
@@ -64,3 +64,16 @@ services:
     depends_on:
       - mqtt
     restart: unless-stopped
+
+  edge-detect:
```
Comment (mchadwick-iqt) [inline docker-compose.yml]: Keep the original docker-compose.yml as-is until we've put the ML components through their paces.  Add this content into a new `docker-compose-edge.yml` for us to do system testing on for a couple weeks to ensure stability before we release this as the formal docker-compose.

Also, this only is primed for one of the two ML models.  Add in both

**Review by mchadwick-iqt — COMMENTED** [review]
One or two other comments - 
  1. Are the TFLite weights in there from base mobilenet, or is this the enhanced ones Daniel did?  I only want the base TFLite weights out there.
  2. There need to be base yolov7 weights for use as well.  These should not be anything involving our collection to start, just a base model weight.
  3. The CI process looks like it will only push one or the other to Dockerhub.  We want both to exist.
  4. At some point soon all of these modules need to be broken out into their own repos (like AISonobuoy).  We can do that now to simplify this work dramatically, or later.  I'm agnostic.

File: edge-detect/coral/base_mqtt_pub_sub.py (hunk 2 above)
Comment (meadej) [reply edge-detect/coral/base_mqtt_pub_sub.py]: Good point. Addressed in https://github.com/IQTLabs/edgetech-coral-detect/pull/1

File: .github/workflows/dockerbuild.yml (hunk 1 above)
Comment (meadej) [reply .github/workflows/dockerbuild.yml]: Addressed in https://github.com/IQTLabs/edgetech-coral-detect/pull/1

File: edge-detect/coral/docker_setup.sh (hunk 4 above)
Comment (meadej) [reply edge-detect/coral/docker_setup.sh]: No, good point. Addressed in https://github.com/IQTLabs/edgetech-coral-detect/pull/1

File: edge-detect/yolo/base_mqtt_pub_sub.py (hunk 7 above)
Comment (meadej) [reply edge-detect/yolo/base_mqtt_pub_sub.py]: Absolutely can. Implemented in https://github.com/IQTLabs/edgetech-coral-detect/pull/1 and https://github.com/IQTLabs/edgetech-yolo-detect/pull/3

File: edge-detect/yolo/Dockerfile.old (hunk 8 above)
Comment (meadej) [reply edge-detect/yolo/Dockerfile.old]: No, unintentional commit. Addressed in https://github.com/IQTLabs/edgetech-coral-detect/pull/1.

File: edge-detect/docker-compose.yml (hunk 6 above)
Comment (meadej) [reply edge-detect/docker-compose.yml]: Removed from https://github.com/IQTLabs/edgetech-coral-detect/pull/1

File: edge-detect/coral/sort_coral.py (hunk 5 above)
Comment (meadej) [reply edge-detect/coral/sort_coral.py]: Addressed in https://github.com/IQTLabs/edgetech-coral-detect/pull/1 and https://github.com/IQTLabs/edgetech-yolo-detect/pull/3

File: edge-detect/yolo/sort.py (hunk 9 above)
Comment (meadej) [reply edge-detect/yolo/sort.py]: Addressed in https://github.com/IQTLabs/edgetech-yolo-detect/pull/3 and https://github.com/IQTLabs/edgetech-coral-detect/pull/1.

File: .github/workflows/dockerbuild.yml (hunk 10 above)
Comment (meadej) [reply .github/workflows/dockerbuild.yml]: Currently publishing to DockerHub as skyscan-yolo and skyscan-coral. Does not appear to be any write-over

File: edge-detect/coral/docker-compose.yml (hunk 3 above)
Comment (meadej) [reply edge-detect/coral/docker-compose.yml]: Renamed paths and clarified this file is demonstrative.

File: docker-compose.yml (hunk 12 above)
Comment (meadej) [reply docker-compose.yml]: Removed from repo. Built new docker compose file.
